### ISSUE TITLE
The field name 'release_date' is confusing, consider changing the name to 'public_date'

### ISSUE THREAD

**Issue by zmanion** [oasis-tcs/csaf#782]
While the documentation is clear, the field name 'release_date' is confusing. Could it be changed to 'public_date' or 'date_public'?

I'd suggest a minor documentation update also: "...the date and time the vulnerability was originally publicly disclosed" (or "published" or "made public").

https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html

3.2.3.11 Vulnerabilities Property - Release Date
Release date (release_date) with value type string of format date-time holds the date and time the vulnerability was originally released into the wild.

3.2.1.12.2 Document Property - Tracking - Current Release Date
Current release date (current_release_date) with value type string with format date-time holds the date when the current revision of this document was released.

3.2.1.12.5 Document Property - Tracking - Initial Release Date
Initial release date (initial_release_date) with value type string with format date-time holds the date when this document was first published.

**Comment by santosomar** [comment]
You bring up an important point about potential confusion with the term. However, it's important to note that CSAF documents are sometimes distributed in a non-public manner and may carry classifications such as TLP amber, red, etc. In these cases, the documents are shared with a restricted audience under specific confidentiality protocols (i.e., TLP).

Using the term `public_date` or `date_public` might imply that the information is always released publicly, which isn't always the case. The term 'release_date' is intentionally used to encompass both public and non-public releases of vulnerability information. It signifies the date and time the vulnerability was initially disclosed, regardless of the audience or distribution limitations.

**Comment by tschmidtb51** [comment]
Regarding the `Vulnerabilities Property - Release Date` - the **public** release is true, for the `Document Property - Tracking - * Release Date` this could also be only available to a closed user group (aka intended audience).

**Comment by zmanion** [comment]
I had not considered a restricted or private "release." But, I think that document/tracking/*_release_date should work for any restricted/TLP level of CSAF distribution.

I'm specifically concerned with the document/vulnerability/release_date, which as defined, seems like it is "date truly public as in on the internet" and which should be true regardless of the TLP. For example, I could deliver a TLP:RED CSAF about a public vulnerability and "date public" would probably be of interest to my readers.

The main issue is just the field name, not the meaning of the field (or any other date fields). It should also be a non-breaking/backwards compatible change.

**Comment by zmanion** [comment]
Whatever the field name, I think it's useful to have a universally applicable datetime stamp for when the vulnerability is published, i.e., make public, on the internet, in the wild. This can be independent of any private notifications, disclosures, or releases.

**Comment by zmanion** [comment]
Also a note that while I was previously an OASIS CSAF member, I was not when I filed this issue. Just ran into this during CSAF use.

**Comment by santosomar** [comment]
During the CSAF TC meeting on 2024-11-27, there was a discussion about this issue (to Differentiate Between `disclosure_date` and `first_known_exploitation_date` in CSAF Documents)

A few folks suggested not to include these fields. However, for the purpose of documentation and awareness. The suggestion is not to include these as mandatory fields. 

It will be good to introduce a differentiation in the `/vulnerabilities[]` section of CSAF documents by defining two distinct fields: `disclosure_date` and `first_known_exploitation_date`. These fields will serve different purposes and follow the guidelines outlined below:  

1. **Disclosure Date (`disclosure_date`):**  
   - **Type:** String of format `date-time`.  
   - **Definition:** Holds the date and time the vulnerability was publicly disclosed.  

2. **Date of First Known Exploitation (`first_known_exploitation_date`):**  
   - **Type:** String of format `date-time`.  
   - **Definition:** Records the date and time the vulnerability was first observed to be exploited in the wild.  This is good for representing CISA KEVs, etc. 
   - **Exclusions:** Does not include exploitation in lab or testing environments.  

**Rationale:**  
- An OPTIONAL field, that clarifies the distinction between when a vulnerability is disclosed and when it is observed to be actively exploited.  

**Action Items:**  
- Send a call to action to the TC about this issue and obtain additional feedback from the TC.

**Comment by mipfu** [comment]
Naming is hard :)

my 2 cents:

Date Timestamp field have the tendency to be used inflationary in any data model. 
Therefore a naming convention is helpful. 
From my personal experience a best practice is 
a <type>_<verb-pastform> convention. 
* publish_date -> datetime_published
* first_known_exploitation_date -> _date_first_known_exploited
* disclosure_date -> date_disclosed

It helps to understand that "from that timestamp value on the status changed to the verbs state. e.g published"

using the present-verb raises the abiquity if it was intended to be published or actually published at that date or was it published.
For the first use datetime_publishing vs datetime_published

**Comment by tschmidtb51** [comment]
After discussing the issue with a few colleagues: We should bind the exploitation to specific products - therefore, I guess a similar structure as `flags` might make sense.

**Comment by rjb4standards** [comment]
Using Log4J as an example, I'm supposing that each product vendor would indicate the "affected status" for each of their products affected by a specific vulnerability. But, what about a generic reporter with no products of their own, should they indicate which products are affected by a vulnerability, even though they are not the authoritative party for a particular product? 

I think it's important to distinguish the "vulnerability disclosure date" from a confirmed "product confirmed affected date" by a product vendor.

**Comment by sthagen** [comment]
The issue was further discussed during the 2024-12-18 TC meeting.

**Comment by sthagen** [comment]
Question: Depending on sharing groups there can be several disclosure dates per vulnerability and product version, right?

**Comment by santosomar** [comment]
As discussed in today's meeting (2024-12-18) and as a follow up to [Stefan's statement above](https://github.com/oasis-tcs/csaf/issues/782#issuecomment-2552050595), we will also create a new issue to address address the second element discussed here (Date of First Known Exploitation (`first_known_exploitation_date`)) in a separate issue.

> The issue was further discussed during the 2024-12-18 TC meeting with focus on part 1 renaming the existing field `release_date` to `disclosure_date`. Omar moved to reinstate the field name for the date a vulnerability was first published from `release_date` to `disclosure_date`, Denny seconded. No discussion, no objections, the motion carries.

**Comment by tschmidtb51** [comment]
- [ ] update schema
- [ ] update prose
- [ ] update guidance on size
- [ ] check whether (and which tests) need updates

**Comment by zmanion** [comment]
https://groups.oasis-open.org/discussion/the-field-name-release-date-is-confusing-consider-changing-the-name-to-public-date-782#bm799a92cb-7232-49d5-9648-ce1a8844764e

**Comment by zmanion** [comment]
> Question: Depending on sharing groups there can be several disclosure date values per vulnerability and product version, right? TLP:RED group X earlier than TLP_CLEAR (later).

I don't think that different disclosure phases-change dates are under consideration here (although a fine idea), my intention was to rename the field that captures the date the vulnerability information is public. For example, I might label vulnerability information TLP:RED, but if it gets published, it's published. "Earliest date found on the internet."

**Comment by tschmidtb51** [comment]
> https://groups.oasis-open.org/discussion/the-field-name-release-date-is-confusing-consider-changing-the-name-to-public-date-782#bm799a92cb-7232-49d5-9648-ce1a8844764e

Thank you @zmanion, for completing the proposal by using the channel OASIS prescribes for external feedback. Much appreciated!

**Comment by tschmidtb51** [comment]
For tracking reasons, the issue will be closed by merging the corresponding PR.

**Comment by tschmidtb51** [comment]
> > Question: Depending on sharing groups there can be several disclosure date values per vulnerability and product version, right? TLP:RED group X earlier than TLP_CLEAR (later).
> 
> I don't think that different disclosure phases-change dates are under consideration here (although a fine idea), my intention was to rename the field that captures the date the vulnerability information is public. For example, I might label vulnerability information TLP:RED, but if it gets published, it's published. "Earliest date found on the internet."

It is now implemented as "publicly released". Any release to a closed group under `>TLP:CLEAR` is not tracked.
